### DISCUSSION TITLE
Generating a PDF places the files in my root folder, is it possible to decide where to place it?

### DISCUSSION THREAD

**Discussion by orgaard** [QuestPDF/QuestPDF#1004] [Q&A] [answered]
I would like to place any generated files in a specific folder, but I can't seem to do that. 

.GeneratePdf($"{fileName}.pdf"); Places the file in the root.

Alternative I could also call GeneratePdfAndShow() but that doesn't give me the option to name the pdf. 

Can anybody point me in the correct direction? 

Thanks

**Comment by MarcinZiabek** [comment, marked as answer]
Please simply provide an absolute or relative path of where the file should be saved 😄 

```csharp
// example:
.GeneratePdf($"/Users/marcinziabek/Downloads/{fileName}.pdf");
```

**Reply by orgaard** [reply]
I did try that, but that gave me a file with the name of the path and the file name. Let me review my code and see if I have made a mistake in that approach.

**Reply by orgaard** [reply]
I was able to achieve what I wanted, almost like I already did, and you suggested. However to make it work, I have to specify the path without the leading "/", so it becomes: 
.GeneratePdf($"Users/marcinziabek/Downloads/{fileName}.pdf");

Is it also possible to name a .pdf file when using .GeneratePdfAndShow()?

**Reply by girlpunk** [reply]
The easiest way to achieve that would be to generate the PDF in the same way you do at the moment, then start a process to have explorer open the PDF at the expected path/filename. For example:

```csharp
document.GeneratePdf(filePath);

using var process = new Process
{
    StartInfo = new ProcessStartInfo(filePath)
    {
        UseShellExecute = true
    }
};

process.Start();
```

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
